Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp

```diff
@@ -432,6 +432,10 @@ void NetworkStorageSession::setCookiesFromDOM(const URL& firstParty, const SameS
         if (!cookie)
             continue;
 
+        auto cookieName = String::fromUTF8(soup_cookie_get_name(cookie.get()));
+        if (cookieName.startsWithIgnoringASCIICase("__Http-"_s) || cookieName.startsWithIgnoringASCIICase("__HostHttp-"_s))
+            continue;
```

Comment:
I'm not familiar with the soup implementation of cookies, so it would good to ensure someone familiar with it approves of this part.